### PR TITLE
BZ2012738: Changing precopy interval

### PR DESCRIPTION
**Pull request by apinnick**
https://bugzilla.redhat.com/show_bug.cgi?id=2012738

Previews:

https://deploy-preview-224--forklift-documentation.netlify.app/documentation/doc-migration_toolkit_for_virtualization/master/#changing-precopy-intervals-ui_forklift

### CODE REVIEW

**Comment by apinnick** [comment]
@istein1  Please review. Thanks.

**Comment by mguetta1** [comment]
@apinnick could you please add how to set the precopy_interval environment variable via UI? 
- Goto Workloads -> Deployments -> 'forklift-controller' -> Environment
- Add env variable named: PRECOPY_INTERVAL and value
- Save

**Comment by apinnick** [comment]
> @apinnick could you please add how to set the precopy_interval environment variable via UI?
> 
>     * Goto Workloads -> Deployments -> 'forklift-controller' -> Environment
> 
>     * Add env variable named: PRECOPY_INTERVAL and value
> 
>     * Save

@mguetta1 Does the forklift-controller pod have to be restarted?

**Comment by mguetta1** [comment]
@apinnick once pressing on 'save' the current forklift-controller pod is terminated and another one is created.
No need to manually restart

**Comment by apinnick** [comment]
@mguetta1 Can you explain the use cases to me? 

Why would someone want to do it with an environmental variable rather than edit the forklift-controller deployment?

**Comment by mguetta1** [comment]
@apinnick this is the same operation, to set the interval time between precopies, via UI.
In both cases, the user edits the forklift-controller deployment by adding an environment variable.
IN my opinion, since we have the ability to set the internal time using UI/CLI (related to MTV 2.2), we should provide both options

File: documentation/modules/changing-precopy-intervals-cli.adoc [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+[id="changing-precopy-intervals-cli_{context}"]
+= Changing precopy intervals for warm migration
+
+The snapshots for the precopy stage of warm migration are created at one-hour intervals by default.
+
+You can change the snapshot interval by editing the `forklift-controller` deployment.
+
+.Prerequisites
+
+* You must enable link:https://kb.vmware.com/s/article/1020128[changed block tracking (CBT)] on the VMware virtual machines and virtual disks.
```
Comment (mguetta1) [inline documentation/modules/changing-precopy-intervals-cli.adoc:14]: This is a general prerequisite in order to be able to warm migrate a VMware VM, not specific to change the interval time between precopies

File: documentation/modules/changing-precopy-intervals-ui.adoc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+[id="changing-precopy-intervals-ui_{context}"]
+= Changing precopy intervals for warm migration
+
+The snapshots for the precopy stage of warm migration are created at one-hour intervals by default.
+
+You can change the snapshot interval by editing the `forklift-controller` deployment in the {ocp} web console.
+
+.Prerequisites
+
+* You must enable link:https://kb.vmware.com/s/article/1020128[changed block tracking (CBT)] on the VMware virtual machines and virtual disks.
```
Comment (mguetta1) [inline documentation/modules/changing-precopy-intervals-ui.adoc:14]: This is a general prerequisite in order to be able to warm migrate a VMware VM, not specific to change the interval time between precopies

**Comment by apinnick** [comment]
> @apinnick this is the same operation, to set the interval time between precopies, via UI. In both cases, the user edits the forklift-controller deployment by adding an environment variable. IN my opinion, since we have the ability to set the internal time using UI/CLI (related to MTV 2.2), we should provide both options

@mguetta1 There needs to be a reason for documenting the same thing twice. 

@fdupont-redhat  Is there a reason why a user might prefer to do it by editing the forklift-controller deployment instead of adding a global variable?

File: documentation/modules/changing-precopy-intervals-cli.adoc (hunk 1 above)
Comment (mguetta1) [reply documentation/modules/changing-precopy-intervals-cli.adoc:14]: oh I see now that there is no dedicated section for warm migration.

Comment (apinnick) [reply documentation/modules/changing-precopy-intervals-cli.adoc:14]: I have not yet decided where this module is going to go.

**Comment by apinnick** [comment]
@mguetta1 I updated the module so that the UI method is documented. See the preview link in the PR description.

I'm still trying to decide whether putting it between "running migration plan" and "canceling migration plan" is the best place.

**Review by mguetta1 — APPROVED** [review]
Thanks @apinnick 
Looks good to me

**Pull request merged**